File: r/integration.r
# Define the limits of integration
a <- -pi
b <- 2*pi/3

# Define the number of intervals
n <- 1000

# Create a vector of x values
x <- seq(a, b, length.out = n)

# Compute the function values
f <- sin(x)

# Compute the area using a Riemann sum
area_riemann <- sum(f * (b - a) / n)

# Print the result
cat("The area between -π and 2/3π for sin(x) is:", area_riemann, "\n")

# Compute the exact value of the integral
exact_result <- -cos(b) + cos(a)

# Print the exact result
cat("The exact area between -π and 2/3π for sin(x) is:", exact_result, "\n")

File: cpp/integration.cpp
#include <iostream>
#include <cmath>

// define the function sin(x)
double f(double x) {
    return sin(x);
}

int main() {
    // define the limits of integration
    double a = -M_PI;
    double b = 2*M_PI/3;

    // define the number of intervals for the Riemann sum
    int n = 1000;

    // compute the width of each interval
    double dx = (b - a) / n;

    // initialize the sum
    double sum = 0;

    // compute the Riemann sum
    for (int i = 0; i < n; i++) {
        double x = a + i * dx;
        sum += f(x) * dx;
    }

    std::cout << "The area between -π and 2/3π for sin(x) is: " << sum << std::endl;

    // compute the exact value of the integral
    double exact_result = -cos(b) + cos(a);

    std::cout << "The exact area between -π and 2/3π for sin(x) is: " << exact_result << std::endl;

    return 0;
}
